### ISSUE TITLE
[FR] Image Grid chart: bigger images, dynamic sizing, and show params in headers

### ISSUE THREAD

**Issue by Rasaboun** [mlflow/mlflow#22117]
> [!WARNING]
> Before submitting a PR, please make sure that:
> - A maintainer has triaged this issue and applied the `ready` label
> - This issue has no assignee
> - No duplicate PR exists
>
> PRs not meeting these requirements may be automatically closed.

### Willingness to contribute

Yes. I can contribute this feature independently.

### Proposal Summary

The Image Grid chart is hard to use when you have many runs and many image keys. Cards are too short (360px), cells use a fixed minimum size that wastes space or overflows, and there's no way to see which hyperparameters each run used without leaving the chart.

I'd like to make the Image Grid actually usable for real training comparison workflows (think LoRA fine-tuning where you compare 10+ configs across 10 sample prompts per step).

### Motivation

> #### What is the use case for this feature?

Comparing generated images across multiple training runs at each step. For example when fine-tuning a LoRA for Stable Diffusion with ai-toolkit, each sampling step produces 10 images (one per prompt). You want to compare those across different LoRA configs (rank 16 vs 32, adamw vs prodigy, different learning rates) to pick the best one.

> #### Why is this use case valuable to support for MLflow users in general?

Anyone doing image generation, style transfer, super-resolution, or any visual ML task needs to compare outputs across runs. The Image Grid chart is the right tool for this but right now it falls short when you have more than 2-3 runs.

> #### Why is this use case valuable to support for your project(s) or organization?

We're building an MLflow integration for ostris/ai-toolkit (LoRA training for diffusion models). Image comparison is the core workflow for evaluating training quality.

> #### Why is it currently difficult to achieve this use case?

Three problems:

1. **Image cards are too short** at 360px. With 10 runs you can barely see the first two images before the card cuts off. You always have to go fullscreen.

2. **The multi-key table uses a fixed minimum cell size (200px)** regardless of how many runs you have. With 10 runs that's 2000px+ of table width so you only see 4-5 runs and have to scroll horizontally to see the rest. The cells should size themselves based on available space.

3. **Column headers only show the run name**. When comparing rank16-adamw vs rank16-prodigy vs rank32-adamw you have to go back to the runs table to remember which hyperparameters each one uses. Showing params directly in the header would save a lot of back and forth.

### Details

I have a working implementation that addresses all three points:

- Image chart cards now use 600px height instead of 360px
- The multi-key table dynamically sizes cells based on container width divided by the number of runs, with a 150px minimum so images stay readable
- Column headers show run params (rank, optimizer, lr etc) below the run name in secondary text, with a tooltip for the full list
- Also bumped MIN_GRID_IMAGE_SIZE from 200 to 400 for single-key cards so individual prompt comparisons show bigger images
- Bonus: made the Compare Runs artifact view use flex layout instead of fixed width columns so images scale properly there too

Files changed are all in `mlflow/server/js/src/experiment-tracking/`:
- `components/runs-charts/components/charts/ImageGridPlot.common.tsx`
- `components/runs-charts/components/charts/ImageGridMultipleKeyPlot.tsx`
- `components/runs-charts/components/charts/ImageGridSingleKeyPlot.tsx`
- `components/runs-charts/components/cards/RunsChartsImageChartCard.tsx`
- `components/CompareRunArtifactView.tsx`
- `components/artifact-view-components/ShowArtifactImageView.tsx`

I have a working implementation and opened a PR alongside this issue so you can review both together. I need this for an MLflow integration I'm building for ostris/ai-toolkit so I went ahead and coded it.

### What machine learning domain(s) is this feature request about?

- `domain/deep-learning`

### What area(s) of MLflow is this feature request about?

- `area/uiux`

## Before

<img width="1200" height="862" alt="Image" src="https://github.com/user-attachments/assets/19e01a79-5fdc-4033-a852-d28bf3658c66" />

## After

<img width="1200" height="862" alt="Image" src="https://github.com/user-attachments/assets/8f5bf086-6bba-4e43-995b-a111996931e2" />

**Comment by TomeHirata** [comment]
Thank you for the FR, the improvements are helpful.

**Comment by Rasaboun** [comment]
@TomeHirata PR is ready https://github.com/mlflow/mlflow/pull/22118

**Comment by Adityax-94** [comment]
is this issue still open. I intend to contribute
